enhancements/csvless-bundles.md

```diff
@@ -24,209 +24,488 @@ status: implementable
+_Ex. Fail to detect mediatype on empty directory_
 
-## Design Details
+```sh
+$ tree .
+.
 
-### Implementation Details
+0 directories, 0 files
+$ opm bundle init .
+# <log: "unable to infer mediatype when no manifests are present">
+```
+
+_Ex. Infer mediatype from src manifests_
+
+```sh
+$ tree kiali-csvless
+kiali-csvless
+├── crds.yaml
+├── deployment.yaml
+└── rbac.yaml
+$ opm bundle init -o kiali kiali-csvless
+# <logs>
+$ tree kiali
+kiali
+├── manifests
+│   ├── crds.yaml
+│   ├── deployments.yaml
+│   └── rbac.yaml
+└── metadata
+    ├── bundle.yaml
+    ├── annotations.yaml
+    └── dependencies.yaml
+
+2 directories, 6 files
+```
+
+Add a new command to `opm`, `opm bundle build`, which takes as input a directory tree -- described in [File Format](#file-format) -- and outputs an Operator Bundle image.
+
+_Ex._
 
-#### Required Manifests
+```sh
+$ opm bundle build . # detect the mediatype of the current directory, build it, and if successful, write the image tar.gz to stdout
+$ opm bundle build -o image://quay.io/my/bundle:v1.0.0 . # detect the mediatype of the current directory, build it, and if successful, push the image to a remote reference
+```
+#### Additively Build an Index
 
-The absence of a CSV means that there is a set of manifests that must ALWAYS be present to build `registry+v2` bundles. These manifests are ALWAYS __required__:
+As a Pipeline Maintainer, I want to additively build an index of `registry+v2` bundles so that I can release new bundles to an existing index.
 
-- a `Deployment` for the operator
-- a `metadata/olm.yaml` file with the following fields
-  - name
-  - version
-  - minKubeVersion
-  - installModes
+##### UX
 
-In certain situations, more manifests are required:
+Extend the `opm index add` command to accept `registry+v2` bundles.
 
-- providing an API from an APIService requires
-  - an `APIService`
-  - the `Service` it references
-- providing an admission webhook
-  - a `MutatingWebhookConfiguration` or a `ValidatingAdmissionWebhook`
-  - the `Service` it references
+#### Direct Application
 
-Other manifests are `optional`:
+As an Operator Author, I want a way to directly apply a `registry+v2` bundle to a cluster -- without a `Subscription` -- so that I can:
 
-- `Roles`, `RoleBindings`, `ClusterRoles`, and `ClusterRolesBindings`
-- `CustomResourceDefinitions`
+- validate the bundle's content
+- rapidly prototype my operator
 
-#### `opm` Support
+##### UX
 
-[`opm`](https://github.com/operator-framework/operator-registry#overview) must be updated to provide support for building `registry+v2` bundles. This can be acheived with explicit flags or through CSV/olm.yaml file presence.
+Add a new command to `opm`, `opm bundle crds`, which takes as input an Operator Bundle and outputs the set of kubectl-able CustomResourceDefinitions (CRDs) it provides.
```

Comment:
Can we have an example here of why this is useful and valuable to users? Is there a reason to separate out the `bundle crds` and `bundle resources` commands from something more generic that just exports the `manifests/` directory explicitly?

Reply:
> Is there a reason to separate out the bundle crds and bundle resources commands from something more generic

probably not, see [this comment](https://github.com/operator-framework/enhancements/pull/29/files/1d082b69f844684bb863f12702f631c894c7514b#r442836809).

>  just exports the manifests/ directory explicitly?

We no longer have CSVs in the `/manifests` directory, so we can't expect OLM to recognize and manage the output without something that performs a transformation.